src/metadata/models/metadata-value.model.ts

```diff
@@ -0,0 +1,40 @@
+/* istanbul ignore file */
+
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { MetadataLockStatus } from '@polymeshassociation/polymesh-sdk/types';
+
+export class MetadataValueModel {
+  @ApiProperty({
+    description: 'Value of the Asset Metadata',
+    type: 'string',
+    example: 'Some metadata',
+  })
+  readonly value: string;
+
+  @ApiProperty({
+    description: 'Date at which the Metadata value expires, null if it never expires',
+    type: 'string',
+    example: new Date('05/23/2021').toISOString(),
+    nullable: true,
+  })
+  readonly expiry: Date | null;
+
+  @ApiProperty({
+    description: 'Lock status of Metadata value',
+    enum: MetadataLockStatus,
+    example: MetadataLockStatus.LockedUntil,
+  })
+  readonly lockStatus: MetadataLockStatus;
+
+  @ApiPropertyOptional({
+    description:
+      'Date till which the Metadata value will be locked. This only applies when `status` is `LockedUntil`',
+    type: 'string',
+    example: new Date('05/23/2021').toISOString(),
+  })
+  readonly lockedUntil?: Date;
```

Comment:
To me this seems a bit inconsistent with `expiry` which is nullable.

I am OK with it as long as you know the difference in behavior and this is what you want: https://dev.to/jsmates/why-the-undefined-variable-is-not-sent-in-a-json-api-response-18k3#:~:text=Undefined%20is%20not%20a%20valid,valid%20JSON%20and%20vice%20versa.&text=All%20other%20values%20will%20be,undefined%20won't%20be%20sent.

Reply:
Wrt to SDK, `null` signifies never in such scenarios (more of like eternity). LockedUntil is only present when status would say `LockedUntil`, hence the `?` part. `null` value here would mean that LockedUntil eternity which isn't the case. So for this scenario it fits correctly IMO